Dockerfile

```diff
@@ -1,32 +1,41 @@
-FROM ocaml/opam:alpine-3.20-ocaml-5.2 as build
+FROM ocaml/opam:alpine-3.20-ocaml-5.2 AS build
 
 # Install system dependencies
-RUN sudo apk update && sudo apk add --update libev-dev openssl-dev gmp-dev oniguruma-dev inotify-tools curl-dev autoconf
+RUN sudo apk -U upgrade --no-cache && sudo apk add --no-cache \
+    autoconf \
+    curl-dev \
+    gmp-dev \
+    inotify-tools \
+    libev-dev \
+    oniguruma-dev \
+    openssl-dev
 
 # Branch freeze was opam-repo HEAD at the time of commit
-RUN cd opam-repository && git pull origin c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
+RUN cd ~/opam-repository && git fetch -q origin master && git reset --hard c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
```

Comment:
```suggest
RUN cd ~/opam-repository && git reset --hard c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
```

That commit is from June 3. Fetching here will is too early. Between two changes of this commit, we want to build the same switch and produce the same docker layer.

Reply:
I'm not sure what you mean…